eternalcore-core/src/main/java/com/eternalcode/core/injector/annotations/component/Component.java

```diff
@@ -0,0 +1,22 @@
+package com.eternalcode.core.injector.annotations.component;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Indicates that the annotated class is a component.
+ * Classes annotated with this annotation will be instantiated by the bean factory and will be available for injection.
+ *
+ * @see Service
+ * @see Repository
+ * @see Task
+ * @see Controller
+ * @see ConfigurationFile
+ * @see BeanSetup
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface Component {
```

Comment:
Correct me if I'm wrong, but if I recall correctly, on last meeting we said that we'll consider removing this class.

Reply:
We can keep this annotation for cases as `ProviderImpl`